cpu/kinetis_common/gpio.c

```diff
@@ -72,304 +183,188 @@ static gpio_state_t config[GPIO_NUMOF];
-            port = GPIO_17_PORT;
-            gpio = GPIO_17_DEV;
-            pin = GPIO_17_PIN;
             break;
 #endif
 #if GPIO_18_EN
 
         case GPIO_18:
             GPIO_18_CLKEN();
-            port = GPIO_18_PORT;
-            gpio = GPIO_18_DEV;
-            pin = GPIO_18_PIN;
             break;
 #endif
 #if GPIO_19_EN
-
         case GPIO_19:
             GPIO_19_CLKEN();
-            port = GPIO_19_PORT;
-            gpio = GPIO_19_DEV;
-            pin = GPIO_19_PIN;
             break;
 #endif
 #if GPIO_20_EN
-
         case GPIO_20:
             GPIO_20_CLKEN();
-            port = GPIO_20_PORT;
-            gpio = GPIO_20_DEV;
-            pin = GPIO_20_PIN;
             break;
 #endif
 #if GPIO_21_EN
-
         case GPIO_21:
             GPIO_21_CLKEN();
-            port = GPIO_21_PORT;
-            gpio = GPIO_21_DEV;
-            pin = GPIO_21_PIN;
             break;
 #endif
 #if GPIO_22_EN
-
         case GPIO_22:
             GPIO_22_CLKEN();
-            port = GPIO_22_PORT;
-            gpio = GPIO_22_DEV;
-            pin = GPIO_22_PIN;
             break;
 #endif
 #if GPIO_23_EN
-
         case GPIO_23:
             GPIO_23_CLKEN();
-            port = GPIO_23_PORT;
-            gpio = GPIO_23_DEV;
-            pin = GPIO_23_PIN;
             break;
 #endif
 #if GPIO_24_EN
-
         case GPIO_24:
             GPIO_24_CLKEN();
-            port = GPIO_24_PORT;
-            gpio = GPIO_24_DEV;
-            pin = GPIO_24_PIN;
             break;
 #endif
 #if GPIO_25_EN
-
         case GPIO_25:
             GPIO_25_CLKEN();
-            port = GPIO_25_PORT;
-            gpio = GPIO_25_DEV;
-            pin = GPIO_25_PIN;
             break;
 #endif
 #if GPIO_26_EN
-
         case GPIO_26:
             GPIO_26_CLKEN();
-            port = GPIO_26_PORT;
-            gpio = GPIO_26_DEV;
-            pin = GPIO_26_PIN;
             break;
 #endif
 #if GPIO_27_EN
-
         case GPIO_27:
             GPIO_27_CLKEN();
-            port = GPIO_27_PORT;
-            gpio = GPIO_27_DEV;
-            pin = GPIO_27_PIN;
             break;
 #endif
 #if GPIO_28_EN
-
         case GPIO_28:
             GPIO_28_CLKEN();
-            port = GPIO_28_PORT;
-            gpio = GPIO_28_DEV;
-            pin = GPIO_28_PIN;
             break;
 #endif
 #if GPIO_29_EN
-
         case GPIO_29:
             GPIO_29_CLKEN();
-            port = GPIO_29_PORT;
-            gpio = GPIO_29_DEV;
-            pin = GPIO_29_PIN;
             break;
 #endif
 #if GPIO_30_EN
-
         case GPIO_30:
             GPIO_30_CLKEN();
-            port = GPIO_30_PORT;
-            gpio = GPIO_30_DEV;
-            pin = GPIO_30_PIN;
             break;
 #endif
 #if GPIO_31_EN
-
         case GPIO_31:
             GPIO_31_CLKEN();
-            port = GPIO_31_PORT;
-            gpio = GPIO_31_DEV;
-            pin = GPIO_31_PIN;
             break;
 #endif
-
         default:
             return -1;
     }
 
+    uint8_t pin = kinetis_gpio_lut[dev].pin;
```

Comment:
for other GPIO implementations (e.g. `cpu/stm32f4/periph/gpio.c`) we always test the given dev parameter to be valid 
```c
if (dev >= GPIO_NUMOF) {
    return -1;
}
```
I would do this here as well.